### PR TITLE
[types] - more complete types for the 3d-tiles spec tiles

### PR DESCRIPTION
**Pull request by dbuck**
While considering a couple options for 3d-tiles extensions for #185 I wanted to fix up some types to track the expected tileset schema for where extensions blobs could be.

* Adds Tileset / TileBase interface to track the 3d-tiles spec schema.
* Adds Tile interface which is exposed in callbacks using during loading, includes specific properties used in overridable functions (Ex: Priority functions).

And: Questionable usefulness without actually using it within the repo ->
* Adds TileInternal type for the internally used tile properties.

### CODE REVIEW

**Comment by dbuck** [comment]
I also updated the referenced three version in the package.json, this is debatable, it shouldn't affect any usage for people using the package directly/normally.

I found I was having type conflicts for types which extend three types when referencing the repo locally using `yarn link` in order to work on the version used in my project directly. Updating the three version in the package.json to a higher rev, after they had split the typedefs out of the main repo, and into @types/three cleared up my local problems.

I chose 128 simply because that's the version we've been using for some number of weeks without any problems ! It could also update to the 134 without any issue, but I've personally not done as much testing against that version yet !

In testing out local examples, everything works as expected except for the `pntsExample.html`, which is trying to reference an example data set from the cesium repo which does not seem to exist anymore? cc: @gkjohnson

File: src/base/TileInternal.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { Tile } from './Tile';
+
+/**
+ * Internal state used/set by the package.
+ */
+
+export interface TileInternal extends Tile {
```
Comment (gkjohnson) [inline src/base/TileInternal.d.ts:7]: We can keep this for now -- might make a good reference? If it looks like it won't be useful we can remove it later.

**Review by gkjohnson — COMMENTED** [review]
Looks good! Just a few changes to make.

**Comment by gkjohnson** [comment]
Excellent! Looks great, thank you. I'll wait to publish since it sounded like you had some other stuff in the works if I recall correctly.

Thanks again!

**Pull request merged**